glados.py

```
@@ -72,6 +72,22 @@ def from_yaml(cls, path: str, key_to_config: Sequence[str] | None = ("Glados",))
+    
+    try:
+        # First attempt with UTF-8
+        with open(path, "r", encoding="utf-8") as file:
+            data = yaml.safe_load(file).get(key, None)
+    except UnicodeDecodeError:
+        # Fall back to utf-8-sig if UTF-8 fails (handles BOM)
+        with open(path, "r", encoding="utf-8-sig") as file:
+            data = yaml.safe_load(file).get(key, None)
+    
+    if data is not None:
+        if "speaker_device" in data.keys():
```

Ruff (0.8.2)

Use a single `if` statement instead of nested `if` statements

Combine `if` statements using `and`

(SIM102)